pennylane/io/qualtran_io.py

```diff
@@ -418,3 +583,403 @@ def compute_matrix(*params, **hyperparams):  # pylint: disable=no-self-argument
+            return "ToBloq(QNode)"
+        if isinstance(self.op, Operation):
+            return f"ToBloq({self.op.name})"
+        return "ToBloq(Qfunc)"
+
+    def __eq__(self, other):
+        if type(other) is type(self):
+            return self.op == other.op
+        return False
+
+    def __hash__(self):
+        return hash(self.op)
+
+    def __str__(self):
+        if hasattr(self.op, "name"):
+            return f"PL{self.op.name}"
+        return "PLQfunc"
+
+
+def to_bloq(circuit, map_ops: bool = True, custom_mapping: dict = None, **kwargs):
+    """
+    Converts a PennyLane circuit or :class:`~.Operation` and returns the representing `Qualtran Bloq <https://qualtran.readthedocs.io/en/latest/bloqs/index.html#bloqs-library>`_.
+
+    .. note::
+        This class requires the latest version of Qualtran. We recommend installing the latest
+        release via ``pip``:
+
+        .. code-block:: console
+
+            pip install qualtran
+
+    Args:
+        circuit (QNode | Operation): a QNode or an initialized PennyLane operator to be wrapped as a Qualtran Bloq.
+        map_ops (bool): Whether or not if the operations are mapped to a Qualtran Bloq or wrapped
+            as a `ToBloq`. Default is True.
+        custom_mapping (dict): Dictionary to specify a mapping between a PennyLane operator and a
+            Qualtran Bloq. A default mapping is used if not defined.
+
+    Returns:
+        Bloq: The Qualtran Bloq that corresponds to the given circuit or :class:`~.Operation` and
+            options.
+
+    .. seealso:: :class:`~.ToBloq`
+
+    **Example**
+
+    This example shows how to use ``qml.to_bloq``:
+
+    .. code-block::
+
+        from qualtran.drawing import show_call_graph, show_counts_sigma
+
+        control_wires = [2, 3]
+        estimation_wires = [4, 5, 6, 7, 8, 9]
+
+        H = -0.4 * qml.Z(0) + 0.3 * qml.Z(1) + 0.4 * qml.Z(0) @ qml.Z(1)
+
+        op = qml.QuantumPhaseEstimation(
+            qml.Qubitization(H, control_wires), estimation_wires=estimation_wires
+        )
+
+        op_as_bloq = qml.to_bloq(op)
+        graph, sigma = wrapped_bloq.call_graph()
+        show_call_graph(graph)
+        show_counts_sigma(sigma)
+
+    .. details::
+        :title: Usage Details
+
+        Not every operator has a direct equivalent in Qualtran. For example, in Qualtran, there
+        exists many varieties of Quantum Phase Estimation. When ``qml.to_bloq`` is called on
+        Quantum Phase Estimation, a smart default is chosen.
+
+        >>> qml.to_bloq(qml.QuantumPhaseEstimation(
+        ...     unitary=qml.RX(0.1, wires=0), estimation_wires=range(1, 5)
+        ... ))
+        TextbookQPE(unitary=Rx(angle=0.1, eps=1e-11), ctrl_state_prep=RectangularWindowState(bitsize=4), qft_inv=Adjoint(subbloq=QFTTextBook(bitsize=4, with_reverse=True)))
+
+        Note that the chosen Qualtran Bloq may not be an exact equivalent. If you want an exact
+        equivalent, we recommend setting ``map_ops`` to False.
```

Comment:
```suggestion
        Note that the chosen Qualtran Bloq may not be an exact equivalent. If an exact
        equivalent is needed, we recommend setting ``map_ops`` to False.
        This will wrap the input PennyLane operator as a Qualtran Bloq, allowing to use Qualtran functions
        such as ``decompose_bloq`` or ``call_graph``.      
```